[Improve][Doc] Add `file_filter_pattern` example to doc

YOMO-Lee opened the pull request:
Added file filtering instructions to the localfile connector documentation

YOMO-Lee committed:
[Fix][Doc] Fix LocalFile doc (#7887)

Supplement and optimize the description of the LocalFile connector on filtering files
[(#7887)](https://github.com/apache/seatunnel/issues/7887)

YOMO-Lee committed:
Merge branch 'apache:dev' into dev

YOMO-Lee committed:
Merge branch 'apache:dev' into dev

YOMO-Lee committed:
[Fix][Connector-V2][ClickHouse] Fix ClickHouse Bug (#7897)

1、When the ClickHouse connector is set to multi parallelism, the task extraction is completed but cannot be stopped normally
[(#7897)](https://github.com/apache/seatunnel/issues/7897)

2、Added E2E test cases for this issue [(#7897)](https://github.com/apache/seatunnel/issues/7897)

3、Local developers want to observe **Job Progress Information** in a timely manner,  Need to modify the following configuration.The configuration in config is invalid
```
seatunnel engine/seatunnel-engineer-common/src/main/resources/seatunnely.yaml
```

YOMO-Lee committed:
[Fix][Connector-V2][ClickHouse] Fix ClickHouse Bug (#7897)

1、When the ClickHouse connector is set to multi parallelism, the task extraction is completed but cannot be stopped normally
[(#7897)](https://github.com/apache/seatunnel/issues/7897)

2、Added E2E test cases for this issue [(#7897)](https://github.com/apache/seatunnel/issues/7897)

3、Local developers want to observe **Job Progress Information** in a timely manner,  Need to modify the following configuration.The configuration in config is invalid
```
seatunnel engine/seatunnel-engineer-common/src/main/resources/seatunnely.yaml
```

YOMO-Lee committed:
[Fix][Connector-V2][ClickHouse] Fix ClickHouse Bug (#7897)

1、When the ClickHouse connector is set to multi parallelism, the task extraction is completed but cannot be stopped normally
[(#7897)](https://github.com/apache/seatunnel/issues/7897)

2、Added E2E test cases for this issue [(#7897)](https://github.com/apache/seatunnel/issues/7897)

3、Local developers want to observe **Job Progress Information** in a timely manner, Need to modify the following configuration.The configuration in config is invalid
```
seatunnel engine/seatunnel-engineer-common/src/main/resources/seatunnely.yaml
```

YOMO-Lee committed:
[Fix][Doc] Fix LocalFile doc (#7887)

Continue to optimize the document about filtering files and add some examples
[(#7887)](https://github.com/apache/seatunnel/issues/7887)

YOMO-Lee committed:
Merge branch 'apache:dev' into dev

YOMO-Lee committed:
[Fix][Doc] Fix LocalFile doc (#7887)

Change to English log
[(#7887)](https://github.com/apache/seatunnel/issues/7887)

YOMO-Lee committed:
Merge remote-tracking branch 'origin/dev' into dev

YOMO-Lee committed:
Merge branch 'apache:dev' into dev

YOMO-Lee committed:
Revert "[Fix][Doc] Fix LocalFile doc (#7887)"

This reverts commit 2e9162d61e5e7074daee1e3a053a795fad8f2d40.

YOMO-Lee committed:
Revert "[Fix][Connector-V2][ClickHouse] Fix ClickHouse Bug (#7897)"

This reverts commit e64b8a67ddd8ae6730813c66cb2e929e2768d6eb.

YOMO-Lee committed:
Revert "[Fix][Connector-V2][ClickHouse] Fix ClickHouse Bug (#7897)"

This reverts commit 1b8066765f2328f4624fb4137b867566f4d2fa69.

YOMO-Lee committed:
Revert "[Fix][Connector-V2][ClickHouse] Fix ClickHouse Bug (#7897)"

This reverts commit 427668128471aea7fb34dcca10666bd079fe79ea.

Hisoka-X commented on docs/en/connector-v2/source/LocalFile.md:
```diff
@@ -254,6 +254,72 @@ Specifies Whether to process data using the tag attribute format.
 
 Filter pattern, which used for filtering files.
 
+The filtering format is similar to wildcard matching file names in Linux.
```
We cannot tell users about the ambiguous conclusion. Please tell users directly that we use Java regular expressions.

Hisoka-X commented on docs/en/connector-v2/source/LocalFile.md:
```diff
@@ -254,6 +254,72 @@ Specifies Whether to process data using the tag attribute format.
 
+The filtering format is similar to wildcard matching file names in Linux.
+
+| Wildcard     | Meaning                                                                                                                        | Example                                                                                                                                      |
+|--------------|--------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
+| *            | Match 0 or more characters                                                                                                     | f* &emsp;&ensp;&emsp; Any file starting with f<br/>b*.txt &emsp; Any file starting with b, any character in the middle, and ending with. txt |
+| []           | Match a single character in parentheses                                                                                        | [abc]* &emsp; A file that starts with any one of the characters a, b, or c                                                                   |
+| ?            | Match any single character                                                                                                     | f?.txt &emsp; Any file starting with 'f' followed by a character and ending with '. txt'                                                     |
+| [!]          | Match any single character not in parentheses                                                                                  | [!abc]* &emsp; Any file that does not start with abc                                                                                         |
+| [a-z]        | Match any single character from a to z                                                                                         | [a-z]* &emsp; Any file starting with a to z                                                                                                  |
+| {a,b,c}/a..z | When separated by commas, it represents individual characters<br/>When separated by two dots, represents continuous characters | {a,b,c}* &emsp; Files starting with any character from abc<br/>{a..Z}* &emsp;&ensp; Files starting with any character from a to z            |
+
+However, it should be noted that unlike Linux wildcard characters, when encountering file suffixes, the middle dot cannot be omitted.
+
+For example, `abc20241022.csv`, the normal Linux wildcard `abc*` is sufficient, but here we need to use `abc*.*` , Pay attention to a point in the middle.
```
Please replace this part to link to https://en.wikipedia.org/wiki/Regular_expression. Let user to learn regular itself.

Hisoka-X commented on docs/en/connector-v2/source/LocalFile.md:
````diff
@@ -254,6 +254,72 @@ Specifies Whether to process data using the tag attribute format.
+
+File Structure Example:
+```
+report.txt
+notes.txt
+input.csv
+abch20241022.csv
+abcw20241022.csv
+abcx20241022.csv
+abcq20241022.csv
+abcg20241022.csv
+abcv20241022.csv
+abcb20241022.csv
+old_data.csv
+logo.png
+script.sh
+helpers.sh
````
Please add some file path, not only match file name.

Hisoka-X reviewed (commented):
Thanks @YOMO-Lee ! I left some comments.

YOMO-Lee committed:
[Fix][Doc] Fix LocalFile Doc

Optimize the describe about Regex

Hisoka-X commented on docs/en/connector-v2/source/LocalFile.md:
```diff
@@ -254,6 +254,54 @@ Specifies Whether to process data using the tag attribute format.
 
 Filter pattern, which used for filtering files.
 
+The pattern follows standard regular expressions. For details, please refer to https://en.wikipedia.org/wiki/Regular_expression. learn it
```
```suggestion
The pattern follows standard regular expressions. For details, please refer to https://en.wikipedia.org/wiki/Regular_expression.
```

Hisoka-X commented on docs/en/connector-v2/source/LocalFile.md:
```diff
@@ -254,6 +254,54 @@ Specifies Whether to process data using the tag attribute format.
 
+The pattern follows standard regular expressions. For details, please refer to https://en.wikipedia.org/wiki/Regular_expression. learn it
+
+File Structure Example:
```
```suggestion
There are some examples.

File Structure Example:
```

YOMO-Lee committed:
[Fix][DOC] LocalFile doc optimize

Optimize document structure

YOMO-Lee committed:
[Fix][DOC] LocalFile doc optimize

Optimize document structure

YOMO-Lee commented:
@Hisoka-X Please review this

YOMO-Lee committed:
[Fix][DOC] Additional explanation for the file_filter_pattern parameter

Please provide a description of all connectors that support the file_filter_pattern parameter

YOMO-Lee committed:
[Fix][DOC] Additional explanation for the file_filter_pattern parameter

Added the following file connector description about file_filter_pattern：
CosFile(en)、OssFile(en)、OssJindoFile(en)、HdfsFile(en)

YOMO-Lee committed:
[Fix][DOC] Additional explanation for the file_filter_pattern parameter

Added the following file connector description about file_filter_pattern：
FtpFile(en)、SftpFile(en)、S3File(en)、HdfsFile(zh)

YOMO-Lee committed:
Merge remote-tracking branch 'origin/LocalFile' into LocalFile

Hisoka-X committed:
update

YOMO-Lee commented:
@zhilinli123 please review
